src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryMath.cs

```diff
@@ -827,33 +827,75 @@ public static FormulaValue Sequence(IRContext irContext, FormulaValue[] args)
+            var rows = LazySequenceDecimal(irContext, count, start.Value, step.Value);
             return new InMemoryTableValue(irContext, StandardTableNodeRecords(irContext, rows.ToArray(), forceSingleColumn: true));
         }
 
-        private static IEnumerable<double> LazySequenceFloat(int records, double start, double step)
+        private static IEnumerable<FormulaValue> LazySequenceFloat(IRContext irContext, int records, double start, double step)
         {
             var x = start;
+            bool isValid = true;
             for (var i = 1; i <= records; i++)
             {
-                yield return x;
-                x += step;
+                if (isValid)
+                {
+                    yield return new NumberValue(IRContext.NotInSource(FormulaType.Number), x);
+                }
+                else
+                {
+                    yield return new ErrorValue(irContext, new ExpressionError()
+                    {
+                        Message = x > 0 ? $"Number is larger than maximum double." : $"Number is less than minimum double.",
+                        Span = irContext.SourceContext,
+                        Kind = ErrorKind.Numeric
+                    });
+                }
+
+                if (isValid && ((double)(x / 2) + (double)(step / 2) <= (double)(double.MaxValue / 2)) && ((double)(x / 2) + (double)(step / 2) >= (double)(double.MinValue / 2)))
+                {
+                    x += step;
+                }
+                else
+                {
+                    isValid = false;
+                }
             }
         }
 
-        private static IEnumerable<decimal> LazySequenceDecimal(int records, decimal start, decimal step)
+        private static IEnumerable<FormulaValue> LazySequenceDecimal(IRContext irContext, int records, decimal start, decimal step)
         {
             var x = start;
+            bool isValid = true;
             for (var i = 1; i <= records; i++)
-            {
-                yield return x;
-                x += step;
+            {                
+                if (isValid)
+                {
+                    yield return new DecimalValue(IRContext.NotInSource(FormulaType.Decimal), x);
+                }
+                else
+                {
+                    yield return new ErrorValue(irContext, new ExpressionError()
+                    {
+                        Message = x > 0 ? $"Number is larger than maximum decimal." : $"Number is less than minimum decimal.",
+                        Span = irContext.SourceContext,
+                        Kind = ErrorKind.Numeric
+                    });
+                }
+
+                if (isValid && ((decimal)(x / 2) + (decimal)(step / 2) <= (decimal)(decimal.MaxValue / 2)) && ((decimal)(x / 2) + (decimal)(step / 2) >= (decimal)(decimal.MinValue / 2)))

```

Comment:
>if (isValid && ((decimal)(x / 2) + (decimal)(step / 2) <= (decimal)(decimal.MaxValue / 2)) && ((decimal)(x / 2) + (decimal)(step / 2) >= (decimal)(decimal.MinValue / 2))) [](http://example.com/codeflow?start=16&length=170)

This check can be simplified by letting C# check the boundaries for you:

    if (isValid) {
        try {
            x += step;
        } catch(OverflowException) {
            isValid = false;
        }
    }

Reply:
updated